tasks/actorstate/actorstatechange.go

```diff
@@ -182,35 +181,15 @@ func (p *ActorStateChangeProcessor) processTipSet(ctx context.Context, node lens
 			return xerrors.Errorf("parse address: %w", err)
 		}
 
-		_, err = node.StateGetActor(ctx, addr, pts.Key())
-		if err != nil {
-			if strings.Contains(err.Error(), "actor not found") {
-				ll.Debugw("actor not found", "addr", str)
-				// TODO consider tracking deleted actors
-				continue
-			}
-			return xerrors.Errorf("get actor: %w", err)
-		}
-
-		_, err = node.StateGetActor(ctx, addr, pts.Parents())
-		if err != nil {
-			if strings.Contains(err.Error(), "actor not found") {
-				ll.Debugw("parent actor not found", "addr", str)
-				// TODO consider tracking deleted actors
-				continue
-			}
-			return xerrors.Errorf("get actor parent: %w", err)
-		}
-
 		palist = append(palist, &visor.ProcessingActor{
 			Head:            act.Head.String(),
 			Code:            act.Code.String(),
 			Nonce:           strconv.FormatUint(act.Nonce, 10),
 			Balance:         act.Balance.String(),
 			Address:         addr.String(),
-			TipSet:          pts.Key().String(),
-			ParentTipSet:    pts.Parents().String(),
-			ParentStateRoot: pts.ParentState().String(),
+			TipSet:          ts.Key().String(),
+			ParentTipSet:    pts.Key().String(),
+			ParentStateRoot: ts.ParentState().String(),
```

Comment:
I believe we have an off-by-one error here:
The actors state changed between `pts` (parent) and `ts` (child) and the stateroot containing the change is the childs stateroot, which is equal to the evaluation of the messages in the parent tipset.

Reply:
I don't think this is right. We are looking at changes between pts.ParentState() (state corresponding to ts.Height()-2) and ts.ParentState() (state corresponding to ts.Height()-1).

The ProcessingActor is getting recorded to process an actor that lives in ts.ParentState(). When processing the actor, the State is obtained with `StateReadState(address, tipset)` where tipset is `pts.Key()`. That is correct, since that corresponds to ts.Height()-1 which was the place where the new values for that actor took place.

At least that is how I understand it, even though this might be unefficient and Lotus has this full mess of what the "state for a tipset" means.